src/ng/directive/ngOptions.js

```diff
@@ -335,6 +335,9 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
           selectValueMap: selectValueMap,
           getOptionFromViewValue: function(value) {
             return selectValueMap[getTrackByValue(value, getLocals(value))];
+          },
+          getViewValueFromOption: function(option) {
+            return trackBy ? angular.copy(option.viewValue) : option.viewValue;
```

Comment:
Since the viewValue might now be an object that could be mutated by the application, we need to make a copy and not return the reference to the value on the option.